xml-support/algorithms.xsd

```diff
@@ -43,19 +45,31 @@
 			</xs:element>
 		</xs:sequence>
 	</xs:complexType>
+	<xs:complexType name="ParamSetWithTokens">
+		<xs:sequence maxOccurs="unbounded">
+			<xs:choice>
+				<xs:element name="param_name" type="xs:token" />
+				<xs:element name="all_of" type="ParamSet" />
+				<xs:element name="one_of" type="ParamSet" />
+				<xs:element name="token" type="Token" />
+			</xs:choice>
+		</xs:sequence>
+	</xs:complexType>
 	<xs:complexType name="ParamSet">
 		<xs:sequence maxOccurs="unbounded">
 			<xs:choice>
-				<xs:element name="param_name" type="xs:token"/>
+				<xs:element name="param_name" type="xs:token" />
 				<xs:element name="all_of" type="ParamSet" />
 				<xs:element name="one_of" type="ParamSet" />
 			</xs:choice>
 		</xs:sequence>
 	</xs:complexType>
 	<xs:complexType name="ResultList">
-		<xs:sequence>
-			<xs:element name="param_type" type="Result"
-				minOccurs="0" maxOccurs="unbounded" />
+		<xs:sequence minOccurs="0" maxOccurs="unbounded">
```

Comment:
I am surprised that the minOccurs is required here.  Does this mean that we have xml documents somewhere that have empty outputs sections?  I ask because I note that the parameters has no such requirement (the question of whether it should is a different one).